### PR TITLE
Add documentation examples for `invalid-all-object`

### PR DESCRIPTION
**Pull request by julthep**
Co-authored-by: Vladyslav Krylasov <vladyslav.krylasov@gmail.com>

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :scroll: Docs          |

## Description
Add documentation examples for `invalid-all-object`

Refs #5953

### CODE REVIEW

File: doc/data/messages/i/invalid-all-object/bad.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+class Fruit:
+    pass
+
+class Worm:
+    pass
+
+__all__ = [Fruit, Worm]  # [invalid-all-object]
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-all-object/bad.py:7]: `__all__` is supposed to be at the top of the file

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Could you add a link to the pep related to `__all__` in details.rst, please ?

File: doc/data/messages/i/invalid-all-object/bad.py (hunk 1 above)
Comment (julthep) [reply doc/data/messages/i/invalid-all-object/bad.py]: Related information has been added to `details.rst` and `related.rst` file. And examples have been revised.

File: doc/data/messages/i/invalid-all-object/bad.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-all-object/bad.py:1]:
```suggestion
__all__ = (
    1, # [invalid-all-object]
    lambda: None, # [invalid-all-object]
    None, # [invalid-all-object],
    Fruit,
    Worm,
)
```

**Review by Pierre-Sassoulas — COMMENTED** [review]
This is an error level message, it's raised for really wrong value like None. Worm instead of "Worm" is bad style but this is not what this check is about. (This is why nothing was raised for bad.py)

File: doc/data/messages/i/invalid-all-object/bad.py [hunk 3]
```diff
@@ -0,0 +1,11 @@
+__all__ = (
+    None, # [invalid-all-object],
```
Comment (Pierre-Sassoulas) [inline doc/data/messages/i/invalid-all-object/bad.py:2]:
```suggestion
    None,  # [invalid-all-object]
```

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**